src/form_components/repeating_group.coffee

```diff
@@ -0,0 +1,155 @@
+FormRenderer.Models.ResponseFieldRepeatingGroup = FormRenderer.Models.BaseFormComponent.extend
+  group: true
+  field_type: 'repeating_group'
+
+  initialize: ->
+    FormRenderer.Models.BaseFormComponent::initialize.apply @, arguments
+    @entries = []
+
+  validateComponent: ->
+    for entry in @entries
+      entry.formComponents.invoke('validateComponent')
+
+  setExistingValue: (entryValues) ->
+    if @isRequired()
+      # If the field is required, ensure that there is at least one value present
+      if !entryValues || entryValues.length == 0
+        entryValues = [{}]
+
+    else # Field is optional...
+      # If entryValues is not set, then add an empty value.
+      if !entryValues
+        entryValues = [{}]
+      # If entryValues is an empty array, the field is skipped.
+      else if _.isArray(entryValues) && _.isEmpty(entryValues)
+        @set('skipped', true)
```

Comment:
When we skip a question, the field value serializes to `[]` since it doesn't contain any fields to answer, like field ID 36 in the repeating sections kitchen sink:

![2017-10-17-204243_484x227_scrot](https://user-images.githubusercontent.com/538235/31682926-c0bc3e04-b37b-11e7-9101-e548b8fc5705.png)

(This is because those nested fields can also be required even if the containing field isn't required, so if you're not skipping the question, so you wouldn't be able to just leave them blank.)

So this if statement is basically saying:

* if we're rendering a form that doesn't already have responses in this repeating group (e.g. `entryValues` is `undefined`, since we're not rendering any data into the nested fields; line 21), create the first internal repeating section (that gets populated with the nested fields). 
* if this question was skipped (e.g. the formData we have for it is `[]`; line 24), we set the `skipped` flag to true so we render the form with it initially skipped, so users can unskip it and answer if they'd like